### DISCUSSION TITLE
关于 l3galley 和 wrapstuff

### DISCUSSION THREAD

**Discussion by zeetex** [qinglee/wrapstuff#7] [General]
wrapstuff 宏包效果很好，我学习了一下代码，发现没用 l3galley 宏包。

l3galley 宏包提供了改变段落形状，在段中挖空等功能。我之前试了一下，效果也很好。

wrapstuff 既然都是用 LaTeX3 语法写的，我想可以试一下 l3galley。

**Comment by qinglee** [comment]
早些年确实有一些用 `l3galley` 实现图文绕排的[例子](https://tex.stackexchange.com/questions/309302/best-way-to-wrap-text-around-figures-in-expl3)。

目前， `l3galley` 还属于 `l3experimental` 目录下，属于实验性质的宏包，已经很多年没有实质性更新了。`l3galley` 重定义了一些广泛使用的命令，会有一定兼容问题。比如，`l3galley` 也重定义了 `\everypar` 和 `\par`，这与当前 LaTeX 段落钩子并不兼容。

虽然 `l3galley` 提供了一些函数用于改变段落形状，但是改变段落形状从来都不是图文绕排这类宏包实现的主要困难。这类宏包的主要困难主要在于易用性和周围环境的兼容性，这方面当前的 `l3galley` 并没有任何帮助，也许将来会纳入 `xor` 框架内考虑。
